### PR TITLE
fix: check if pyproject has poetry earlier

### LINKED ISSUE #400: Poetry
This is related to #324 
When I have a `pyproject.toml` file I must explicitly say `usePoetry: false` to package the requirements.

In the current versions of things I get `ENOENT: no such file or directory, open '/Users/kyle/projects/serverless-python-requirements-bug/.serverless/requirements.txt'`

The above error results from falling into https://github.com/UnitedIncome/serverless-python-requirements/blob/c9d910cc40d95163f8894abfdefaf9c4fc33b505/lib/pip.js#L66 

Whatever the logic is to determine if `usePoetry` is true seems to be at fault.

A working example is at https://github.com/khornberg/serverless-python-requirements-bug

- Clone and run `sls package` to get the error
- Uncomment the `custom` section in the `serverless.yaml` file. Running `sls package` again works.

There is a work around but it seems unintuitive

### PR DESCRIPTION
**Pull request by miketheman**
During the evaluation in the package phase, we determine whether a
`requirements.txt` file exists, or whether we need to generate one.

Since the `pyproject.toml` file is used by poetry, but only if a stanza
is contained inside the file, use the function `isPoetryProject()` along
with the configuration value, thereby reducing the need for a project to
have to declare a configuration override.

Refs #324
Refs #344
Fixes #400

Signed-off-by: Mike Fiedler <miketheman@gmail.com>

### CODE REVIEW

**Comment by miketheman** [comment]
Will repair tonight/tomorrow. This is a classic case of splitting up the ci job to workflows - get that fast feedback on the lint before the beefy test suite.

File: lib/pip.js [hunk 1]
```diff
@@ -62,7 +63,9 @@ function generateRequirementsFile(
 ) {
   if (
     options.usePoetry &&
-    fse.existsSync(path.join(servicePath, 'pyproject.toml'))
+    fse.existsSync(
+      path.join(servicePath, 'pyproject.toml') && isPoetryProject(servicePath)
```
Comment (revmischa) [inline lib/pip.js]: I believe this breaks poetry if it is actually in usee
Shouldn't this be:
`fse.existsSync(path.join(servicePath, 'pyproject.toml')) && isPoetryProject(servicePath)`
not 
`fse.existsSync(path.join(servicePath, 'pyproject.toml') && isPoetryProject(servicePath))`

and same below?

causing this issue: https://github.com/UnitedIncome/serverless-python-requirements/issues/497

Comment (bsamuel-ui) [reply lib/pip.js]: Parentheses are tricky little buggers.

Comment (bsamuel-ui) [reply lib/pip.js]: (I'm going to fix this and merge.)

**Pull request merged**